packages/server/src/fhir/operations/agentpush.ts

```diff
@@ -35,45 +36,78 @@ const MAX_WAIT_TIMEOUT = 60000;
  * Returns the outcome of the agent execution.
  */
 export const agentPushHandler = asyncWrap(async (req: Request, res: Response) => {
+  if (req.header('Prefer') === 'respond-async') {
```

Comment:
It's not immediately obvious to me, but I suspect there's a pattern that we could develop

```ts
function maybeAsync(req: Request, res: Response, handler: SomeFunctionType): void
if (isPreferAsync(req)) {
  sendAsyncResponse(req, res, handler);
} else {
  sendResponse(handler(req, res));
}
```

+/- `async`, `await`, and a bunch of error handling and whatnot
 
With the long term goal that 100% of FHIR operations could automatically get this exact same behavior.

I'm not totally sure if we have all the right primitives and utilities to make that happen, so I'm totally ok to ship this, and let the patterns bake a little bit more.